64 bit targets often don't have a separate fcntl64() system call, because they don't need one.

Signed-off-by: Rob Landley <rob at landley.net>

--- uClibc/include/fcntl.h      2010-04-02 10:34:27.000000000 -0500
+++ uClibc.bak/include/fcntl.h  2010-05-16 04:54:10.000000000 -0500
@@ -73,7 +73,7 @@
 
    This function is a cancellation point and therefore not marked with
    __THROW.  */
-#ifndef __USE_FILE_OFFSET64
+#if !defined(__USE_FILE_OFFSET64) || defined(__LP64__)
 extern int fcntl (int __fd, int __cmd, ...);
 libc_hidden_proto(fcntl)
 #else
@@ -83,7 +83,7 @@
 #  define fcntl fcntl64
 # endif
 #endif
-#ifdef __USE_LARGEFILE64
+#if defined(__USE_LARGEFILE64) && !defined(__LP64__)
 extern int fcntl64 (int __fd, int __cmd, ...);
 libc_hidden_proto(fcntl64)
 #endif

--
